### PR TITLE
Big bag of types

### PR DESCRIPTION
**Pull request by jamieparkinson**
Sorry, I got a bit carried away with pulling the thread on this. It converts the following to typescript:

- The works/images fetching services
- The whole IIIF viewer and related bits
- The archive tree

### CODE REVIEW

File: catalogue/webapp/components/MultipleManifestList/MultipleManifestList.tsx [hunk 1]
```diff
@@ -47,45 +46,34 @@ const HiddenContent = styled.div.attrs(props => ({
...
+    }
+  };
   useEffect(() => {
     window.document.addEventListener('click', handleClickOutside, false);
     return () => {
       window.document.removeEventListener('click', handleClickOutside, false);
     };
   }, []);
 
-  const handleClickOutside = event => {
```
Comment (jamieparkinson) [inline catalogue/webapp/components/MultipleManifestList/MultipleManifestList.tsx:73]: Does anyone know why this block exists? afaict the `downloadText` ref doesn't and has never been attached to anything

File: common/views/components/IIIFViewer/IIIFViewer.tsx [hunk 2]
```diff
@@ -217,22 +208,27 @@ const IIIFViewerComponent = ({
...
+  >();
   const [activeIndex, setActiveIndex] = useState(0);
   const [pageHeight, setPageHeight] = useState(500);
   const [pageWidth, setPageWidth] = useState(1000);
   const [showZoomed, setShowZoomed] = useState(false);
-  const [zoomInfoUrl, setZoomInfoUrl] = useState(null);
-  const [rotatedImages, setRotatedImages] = useState([]);
+  const [zoomInfoUrl, setZoomInfoUrl] = useState<string | undefined>();
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const [rotatedImages, setRotatedImages] = useState<any[]>([]);
```
Comment (jamieparkinson) [inline common/views/components/IIIFViewer/IIIFViewer.tsx:223]: Sorry, I couldn't work out what this or `imageJson` were supposed to be and was in deep at this point

File: common/test/utils/works.test.ts [hunk 3]
```diff
@@ -63,6 +61,7 @@ describe('getItemIdentifiersWith', () => {
 describe('getArchiveAncestorArray', () => {
   it('gets the ancestors of an archive work', () => {
     const archiveAncestorArray = getArchiveAncestorArray(workWithPartOf);
+    console.log(archiveAncestorArray);
```
Comment (kenoir) [inline common/test/utils/works.test.ts:64]: `log`

Comment (jamieparkinson) [reply common/test/utils/works.test.ts:64]: oop

Comment (jamieparkinson) [reply common/test/utils/works.test.ts:64]: debugging à l'ancienne

File: catalogue/webapp/components/MultipleManifestList/MultipleManifestList.tsx (hunk 1 above)
Comment (jamesgorrie) [reply catalogue/webapp/components/MultipleManifestList/MultipleManifestList.tsx:73]: Ping @gestchild - but as far as I can see - I'd agree.

**Review by jamesgorrie — APPROVED** [review]

**Comment by jamesgorrie** [comment]
@jamieparkinson it might be good to try get these changes in in smaller chunk, although I know it's not always possible.

**Pull request merged**